### PR TITLE
Adding a URLAnalyzer and PastebinURLAnalyzer to address issue #14

### PR DESCRIPTION
**Pull request by heavysaturn**
Also added an optional `resolve` constructor arg, which if set to True will attempt to validate the URL it matches by resolving it with urllib.

This PR addresses issue #14

### CODE REVIEW

**Comment by d-Rickyy-b** [comment]
Thank you very much for your contribution. I will take a closer look at it when at home. Looks very nice at first glance.

**Comment by heavysaturn** [comment]
no problem. your hacktoberfest tags brought me here, fun project!

File: analyzers/urlanalyzer.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    be resolved.
+    """
+    name = "URLAnalyzer"
+
+    def __init__(self, action, regex, resolve=False):
+        super().__init__(action, regex)
+        self.regex = re.compile(regex)
+        self.resolve = resolve  # Should we try to resolve the URLs?
+
+    def _resolve_url(self, url):
```
Comment (d-Rickyy-b) [inline analyzers/urlanalyzer.py]: I love the possibility to try to resolve urls. This is a really useful use-case I have not yet thought about. Well done.

File: analyzers/urlanalyzer.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+import re
+import urllib.request
+import urllib.error
+
+from .basicanalyzer import BasicAnalyzer
+
+
+class URLAnalyzer(BasicAnalyzer):
```
Comment (d-Rickyy-b) [inline analyzers/urlanalyzer.py]: I love it. Well done!

File: analyzers/pastebinurlanalyzer.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+from .urlanalyzer import URLAnalyzer
+
+
+class PastebinURLAnalyzer(URLAnalyzer):
+    """ Analyzer to match Pastebin URLs via regex """
+    name = "PastebinURLAnalyzer"
+
+    def __init__(self, action, resolve=False):
+        regex = "((?:https?:\/\/)?pastebin.com\/\S+)"
+        super().__init__(action, regex, resolve)
```
Comment (d-Rickyy-b) [inline analyzers/pastebinurlanalyzer.py]: Missing newline, but I'll add it later :)

Also nice that you subclassed URL Analyzer as written in the issue.

File: analyzers/creditcardanalyzer.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
-from analyzers import RegexAnalyzer
+# -*- coding: utf-8 -*-
```
Comment (d-Rickyy-b) [inline analyzers/creditcardanalyzer.py]: Was missing. Good catch!

**Comment by d-Rickyy-b** [comment]
> 
> 
> no problem. your hacktoberfest tags brought me here, fun project!

Thanks :) Glad you found and like it. I am going to merge your PR. Great work, thanks again.

**Pull request merged**